src/public/modules/forms/helpers/decryption.worker.js

```diff
@@ -150,10 +152,27 @@ async function decryptIntoCsv(data) {
           }
         })
 
-        downloadBlob = await downloadAndDecryptAttachmentsAsZip(
-          attachmentDownloadUrls,
-          secretKey,
-        )
+        let downloadStatus = {
+          _id: '000000000000000000000000',
+          fieldType: 'textfield',
+          question: 'Attachment Download Status',
+          answer: 'Unknown',
+          questionNumber: 1000000,
+        }
+
+        try {
+          downloadBlob = await queue.add(() =>
+            downloadAndDecryptAttachmentsAsZip(
+              attachmentDownloadUrls,
+              secretKey,
+            ),
+          )
+          downloadStatus.answer = 'Success'
+        } catch (error) {
+          csvRecord.status = 'ATTACHMENT_ERROR'
+          downloadStatus.answer = 'Failed'
+        }
+        csvRecord.submissionData.record.unshift(downloadStatus)
```

Comment:
is it possible that `record` is uninitialised at this point, causing the worker to crash?

Reply:
True, resolved with 93d52fe